gateway/blocks_backend.go

```diff
@@ -508,6 +508,9 @@ func walkGatewaySimpleSelector(ctx context.Context, p path.ImmutablePath, params
 					return err
 				}
 				from = fileLength + entityRange.From
+				if from < 0 {
+					from = 0
+				}
```

Comment:
@rvagg @hannahhoward adding a guard here makes sense, but implicit fixup to 0 makes me a bit uneasy.

If the `fileLength` is 100, and user explicitly requested `entity-bytes=-1000:-75`, 
should we ignore the fact the user expects to way more than 25?

We've been trying to avoid surprises in Gateway semantics, and follow conventions from HTTP RFCs where possible. When designing `entity-bytes`,  we've been trying to follow HTTP Range Requests semantics, and those do not silently fixup ranges for the user, but return HTTP Error 416 Range Not Satisfiable every time things are out of range, removing surface for hard to debug errors and allowing user to inspect situation and retry with correct range.


Could we return below error instead?

```suggestion
				from = fileLength + entityRange.From
				if from < 0 {
					return fmt.Errorf("tried to read beyond the file length from the end")
				}
```

Reply:
Sorry for failing to engage -- I'm fine with either outcome. Though I should note if you do this one of your gateway tests uses a From with a very negative value -- the very one I corrected. It will need to be changed to produce an error, and this will put it out of alignment with the IPLD prime implementation.

Reply:
IOW, that means that either this test (the one I submitted the Gateway Conformance PR for) should be removed entirely, or erroring should be added to the actual trustless spec.

Personally, I don't think an obscure edge case is important enough to warrent being in the spec, so personally I'd just remove that gateway conformance test.

Reply:
cc: @rvagg @lidel 

Also, I'm pretty immersed in Saturn stuff FYI so just to say my engagement may be a bit slow.